core/capabilities/ccip/oraclecreator/bootstrap.go

```diff
@@ -104,5 +133,241 @@ func (i *bootstrapOracleCreator) Create(_ uint32, config cctypes.OCR3ConfigWithM
+
+	ctx, cf := context.WithCancel(context.Background())
+	d.syncCtxCf = cf
+
+	go func() {
+		d.sync()
+
+		syncTicker := time.NewTicker(d.syncInterval)
+		for {
+			select {
+			case <-syncTicker.C:
+				d.sync()
+			case <-ctx.Done():
+				return
+			}
+		}
+	}()
+}
+
+func (d *peerGroupDialer) Close() error {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	d.closeExistingPeerGroups()
+
+	if d.syncCtxCf != nil {
+		d.syncCtxCf()
+	}
+
+	return nil
+}
+
+func (d *peerGroupDialer) sync() {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	if !d.shouldSync() {
+		d.lggr.Debugw("No need to sync peer groups")
+		return
+	}
+	d.lggr.Infow("Syncing peer groups")
+
+	d.closeExistingPeerGroups()
+
+	if err := d.createNewPeerGroups(); err != nil {
+		d.lggr.Errorw("failed to create new peer groups", "err", err)
+		d.closeExistingPeerGroups() // close potentially opened peer groups
+	}
+}
+
+func (d *peerGroupDialer) shouldSync() bool {
+	if len(d.activePeerGroups) == 0 {
+		return true
+	}
+
+	activeConfigDigest, candidateConfigDigest := d.rmnHomeReader.GetAllConfigDigests()
+	var configDigests [][32]byte
+
+	if !activeConfigDigest.IsEmpty() {
+		configDigests = append(configDigests, activeConfigDigest)
+	}
+	if !candidateConfigDigest.IsEmpty() {
+		configDigests = append(configDigests, candidateConfigDigest)
+	}
+
+	if len(configDigests) != len(d.activeConfigDigests) {
+		return true
+	}
+	for i, rmnHomeConfigDigest := range configDigests {
+		if rmnHomeConfigDigest != d.activeConfigDigests[i] {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (d *peerGroupDialer) closeExistingPeerGroups() {
+	for _, pg := range d.activePeerGroups {
+		if err := pg.Close(); err != nil {
+			d.lggr.Warnw("failed to close peer group", "err", err)
+			continue
+		}
+		d.lggr.Infow("Closed peer group successfully")
+	}
+
+	d.activePeerGroups = []rmn.PeerGroup{}
+	d.activeConfigDigests = []cciptypes.Bytes32{}
+}
+
+func (d *peerGroupDialer) createNewPeerGroups() error {
+	activeConfigDigest, candidateConfigDigest := d.rmnHomeReader.GetAllConfigDigests()
+	var configDigests [][32]byte
+
+	if !activeConfigDigest.IsEmpty() {
+		configDigests = append(configDigests, activeConfigDigest)
+	}
+	if !candidateConfigDigest.IsEmpty() {
+		configDigests = append(configDigests, candidateConfigDigest)
+	}
+
+	d.lggr.Infow("Creating new peer groups", "configDigests", configDigests)
+
+	for _, rmnHomeConfigDigest := range configDigests {
+		rmnNodesInfo, err := d.rmnHomeReader.GetRMNNodesInfo(rmnHomeConfigDigest)
+		if err != nil {
+			return fmt.Errorf("get RMN nodes info: %w", err)
+		}
+
+		h := sha256.Sum256(append(d.commitConfigDigest[:], rmnHomeConfigDigest[:]...))
+		genericEndpointConfigDigest := writePrefix(ocr2types.ConfigDigestPrefixCCIPMultiRoleRMNCombo, h)
+
+		peerIDs := make([]string, 0, len(d.oraclePeerIDs))
+		for _, p := range d.oraclePeerIDs {
+			peerIDs = append(peerIDs, p.String())
+		}
+		for _, n := range rmnNodesInfo {
+			peerIDs = append(peerIDs, n.PeerID.String())
+		}
```

Comment:
Lets pull this code into a common module, perhaps chainlink-ccip, and lets use it in both this component and the plugin wherever NewPeerGroup is called